errors.go

```diff
@@ -0,0 +1,248 @@
+package gobayeux
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	// ErrClientNotConnected is returned when the client is not connected
+	ErrClientNotConnected = errors.New("client not connected to server")
+
+	// ErrTooManyMessages is returned when there is more than one handshake message
+	ErrTooManyMessages = errors.New("more messages than expected in handshake response")
+
+	// ErrBadChannel is returned when the handshake response is on the wrong channel
+	ErrBadChannel = errors.New("handshake responses must come back via the /meta/handshake channel")
+
+	// ErrFailedToConnect is a general connection error
+	ErrFailedToConnect = errors.New("connect request was not successful")
+
+	// ErrNoSupportedConnectionTypes is returned when the client and server
+	// aren't able to agree on a connection type
+	ErrNoSupportedConnectionTypes = errors.New("no supported connection types provided")
+
+	// ErrNoVersion is returned when a version is not provided
+	ErrNoVersion = errors.New("no version specified")
+
+	// ErrMissingClientID is returned when the client id has not been set
+	ErrMissingClientID = errors.New("missing clientID value")
+
+	// ErrMissingConnectionType is returned when the connection type is unset
+	ErrMissingConnectionType = errors.New("missing connectionType value")
+)
+
+// ErrConnectionFailed is returned whenever Connect is called and it fails
+type ErrConnectionFailed struct {
+	err error
+}
+
+func (e ErrConnectionFailed) Error() string {
+	return fmt.Sprintf("connection failed (%s)", e.err)
+}
+
+func (e ErrConnectionFailed) Unwrap() error {
+	return e.err
+}
+
+// ErrHandshakeFailed is returned whenever the handshake fails
+type ErrHandshakeFailed struct {
+	err error
+}
+
+func (e ErrHandshakeFailed) Error() string {
+	return e.err.Error()
+}
+
+func (e ErrHandshakeFailed) Unwrap() error {
+	return e.err
+}
+
+func newHandshakeError(msg string) *ErrHandshakeFailed {
+	return &ErrHandshakeFailed{
+		fmt.Errorf("handshake was not successful: %s", msg),
+	}
+}
+
+// ErrSubscriptionFailed is returned for any errors on Subscribe
+type ErrSubscriptionFailed struct {
+	Channels []Channel
+	err      error
+}
+
+func (e ErrSubscriptionFailed) Error() string {
+	return fmt.Sprintf("subscription failed (%s)", e.err)
+}
+
+func (e ErrSubscriptionFailed) Unwrap() error {
+	return e.err
+}
+
+// ErrUnsubscribeFailed is returned for any errors on Unsubscribe
+type ErrUnsubscribeFailed struct {
+	Channels []Channel
+	err      error
+}
+
+func (e ErrUnsubscribeFailed) Error() string {
+	return fmt.Sprintf("subscription failed (%s)", e.err)
+}
+
+func (e ErrUnsubscribeFailed) Unwrap() error {
+	return e.err
+}
+
+// ErrActionFailed is a general purpose error returned by the BayeuxClient
+type ErrActionFailed struct {
+	action string
```

Comment:
Why not make action public for example? People otherwise have to parse it themselves